scantools/run_qrcode_detection.py

```diff
@@ -0,0 +1,222 @@
+import argparse
+import multiprocessing
+from pathlib import Path
+from typing import List, Optional
+
+from scantools import (
+    logger,
+    run_meshing,
+    run_navvis_to_capture,
+    to_meshlab_visualization,
+)
+from scantools.capture import Capture
+from scantools.proc.qrcode.detector import QRCodeDetector
+from scantools.proc.qrcode.map import (
+    create_qr_map,
+    filter_qr_codes_by_area,
+    save_qr_maps,
+)
+
+description = """
+This script is used for QR code detection. The script saves the QR map (list of
+all QR codes) in the `CAPTURE_PATH/proc/qrcodes/` directory. The QR map can be
+saved in both TXT and JSON formats. The TXT format is used for compatibility
+with the rest of CAPTURE format. The JSON format is used for convince but
+disabled by default. The script also saves the detected QR codes for each image.
+These are saved in the `CAPTURE_PATH/proc/qrcodes/` directory with the same name
+as the image file, but with the `.qrcodes.txt` suffix. These files are used to
+load the QR codes if they have already been detected for the image. This is
+useful when we want to re-run the script without having to detect the QR codes
+again.
+"""
+
+
+def run(
+    capture_path: Path,
+    navvis_path: Optional[Path] = None,
+    sessions: Optional[List[str]] = None,
+    visualization: bool = True,
+    **kargs,
+):
+    if capture_path.exists():
+        capture = Capture.load(capture_path)
+    else:
+        capture = Capture(sessions={}, path=capture_path)
+
+        logger.info("Convert NavVis data to CAPTURE format: ", navvis_path)
+        # Typically, QR codes are captured intentionally by approaching them
+        # closely. As a result, we don't need tiling, which could potentially
+        # split the QR code across multiple tiles.
+        tiles_format = "none"
+        run_navvis_to_capture.run(navvis_path, capture, tiles_format)
+
+    # If `sessions` is not provided, run for all sessions in the `capture_path`.
+    if sessions is None:
+        sessions = capture.sessions.keys()
+
+    mesh_id = "mesh"
```

Comment:
Maybe use the simplified since raybender will require a lot of RAM for the full one?